ToDoList/src/controller/ToDoList.java

```diff
@@ -0,0 +1,87 @@
+package controller;
+
+import java.io.File;
+
+/**
+ * 
+ * This interface provides a way to communicate between the GUI and the
+ * actual model without the two knowing about the other's implementation
+ * 
+ * This will enable easier testing, as there is less coupling
+ * 
+ * @author Krishna Sannasi
+ *
+ */
+public interface ToDoList
+{
+    /**
+     * Inserts an element into the ToDo list, 
+     * 
+     * if the priority of <code>e</code> conflicts with the priority of an element
+     * already in the list, then it should bump down the priority of the other item
+     * and insert this element into the list
+     * 
+     * @param e - the element to be inserted
+     */
+    void insertElement(Element e);
+    
+    /**
+     * Removes the element equal the the given element
+     * 
+     * @param e
+     * @return - true if the element was successfully removed, false otherwise
+     */
+    boolean removeElement(Element e);
+    
+    /**
+     * Sorts by the given field
+     * 
+     * @param s - the field to sort by
+     */
+    void sortBy(Sort s);
+    
+    /**
+     * Moves an element to the given index
+     * 
+     * @param e - element to move
+     * @param index - index to move to
+     * @return - true if the element was successfully moved, false otherwise
+     */
+    boolean moveElementToIndex(Element e, int index);
+    
+    /**
+     * Gets the element in the list that is equal to the given element
+     * 
+     * @param e
+     * @return
+     */
+    Element getElement(Element e);

```

Comment:
I am unsure how we will be accessing elements. My understanding is that we will be placing them into an array (by insertion of tasks). So we have an array of elements, and these elements have fields (i.e. status, name, priority etc.). However if we want to access an element, the only way we know the difference between them is by the content of their field values. Since the elements are in an array, if we wanted to access an element, we would just access the element directly with array indexing (we could find the index by searching the array of objects based on a particular field). To sum it up, I am unsure what this function does, or more specifically how it would be called in practice. I have a feeling i'm overlooking something obvious, so I hope you can help me see what I'm missing.

Reply:
Yeah, I'm not sure, we could change this to take a priority instead of an element or sonething else.